synfig-studio/src/gui/canvasview.cpp

```diff
@@ -2701,6 +2706,14 @@ CanvasView::set_onion_skins()
 	work_area->set_onion_skins(onion_skins);
 }
 
+void
+CanvasView::toggle_show_ruler()
+{
+	work_area->toggle_ruler();
```

Comment:
Here you can use `work_area->set_ruler_visible(set_ruler_show_toggle...)` and remove `get_ruler_status` and `toggle_ruler` as they are no longer needed.

Reply:
sounds good yes thanks again (^_^).

Reply:
> Here you can use `work_area->set_ruler_visible(set_ruler_show_toggle...)` and remove `get_ruler_status` and `toggle_ruler` as they are no longer needed.

ok @ice0 , I'm a bit sleepy here so maybe I'm missing something😴, but why is ruler status not needed anymore? It is the flag used for saving the metadata, and get_ruler_status() is its getter because here in canvasview.cpp we are outside the class. Maybe you mean to deal with the meta data here right away? but isn't it better to leave it as it is there as all the other related meta data are saved the same way?

Reply:
The flag is already saved inside `set_ruler_show_toggle` widget.
When you call `toggle_show_ruler` you need to toggle this status and just set the visibility for the rulers.
Check this at the morning maybe you'll understand better what I mean.

Reply:
> It is the flag used for saving the metadata

Got it. I need to test some ideas in this case.

Reply:
> > It is the flag used for saving the metadata
> 
> Got it. I need to test some ideas in this case.

sounds good, please let me know when you reach something so we can get this PR also out of the way.